cirq-core/cirq/ops/pauli_string_phasor.py

```diff
@@ -58,26 +60,62 @@ def __init__(
                 in the form of the t in (-1)**t = exp(i pi t).
 
         Raises:
-            ValueError: If the given pauli string does not have eignevalues +1
-                or -1.
+            ValueError: If coefficient is not 1 or -1.
         """
-        if pauli_string.coefficient == -1:
-            pauli_string = -pauli_string
-            exponent_pos, exponent_neg = exponent_neg, exponent_pos
+        gate = PauliStringPhasorGate(
+            dps.DensePauliString(pauli_string.values(), coefficient=pauli_string.coefficient),
+            exponent_neg=exponent_neg,
+            exponent_pos=exponent_pos,
+        )
+        super().__init__(gate, pauli_string.qubits)
+        self._pauli_gate = gate
+        self._pauli_string = gate.dense_pauli_string.on(*self.qubits)
 
-        if pauli_string.coefficient != 1:
-            raise ValueError(
-                "Given PauliString doesn't have +1 and -1 eigenvalues. "
-                "pauli_string.coefficient must be 1 or -1."
-            )
+    @property
+    def exponent_neg(self):
+        return self._pauli_gate.exponent_neg
+
+    @exponent_neg.setter  # type: ignore
+    @deprecated(
+        deadline="v0.15",
+        fix="The mutators of this class are deprecated, instantiate a new object instead.",
+    )
+    def exponent_neg(self, exponent_neg):
+        # coverage: ignore
+        self._pauli_gate._exponent_neg = exponent_neg
+
+    @property
+    def exponent_pos(self):
+        return self._pauli_gate.exponent_pos
+
+    @exponent_pos.setter  # type: ignore
+    @deprecated(
+        deadline="v0.15",
+        fix="The mutators of this class are deprecated, instantiate a new object instead.",
+    )
+    def exponent_pos(self, exponent_pos):
+        # coverage: ignore
+        self._pauli_gate._exponent_pos = exponent_pos
 
-        super().__init__(pauli_string)
-        self.exponent_neg = value.canonicalize_half_turns(exponent_neg)
-        self.exponent_pos = value.canonicalize_half_turns(exponent_pos)
+    @property
+    def pauli_string(self):
+        return self._pauli_string
+
+    @pauli_string.setter  # type: ignore
+    @deprecated(
+        deadline="v0.15",
+        fix="The mutators of this class are deprecated, instantiate a new object instead.",
+    )
+    def pauli_string(self, pauli_string):
+        # coverage: ignore
+        self._pauli_string = pauli_string
+        self._pauli_gate._dense_pauli_string = dps.DensePauliString(
+            pauli_string.values(), coefficient=pauli_string.coefficient
+        )
 
     @property
     def exponent_relative(self) -> Union[int, float, sympy.Basic]:
-        return value.canonicalize_half_turns(self.exponent_neg - self.exponent_pos)
+        return self._pauli_gate.exponent_relative
 
     def _value_equality_values_(self):
```

Comment:
`_value_equality_values_` not needed as it'll be derived from `GateOperation` and `self.pauli_string` is redundant if we consider `self.gate` and `self.qubits`, which would automatically get considered in `GateOperation`

Reply:
We'd have to define a `qubit_index_to_equivalence_group_key` on the `PauliStringPhasorGate` for the implicit equality to work correctly. I leave the choice up to you on whether to use the implicit version or leave it as is.

Reply:
Since PauliStringPhasor itself is likely to be removed at some point, I'll leave the additional complexity there.

Reply:
Or actually this is going to have to be done regardless if we move to GateOperation, so I'll implement that function.

Reply:
This didn't work for me, as it still returned not equal. Reverting to the existing method.

Reply:
I see, interchangeable qubits isn't enough here. `X(0)*Y(1)` ~ `Y(1)*X(0)`, but you can't just interchange qubits you have to interchange the paulis too. Interchangeable qubits is sufficient for single-pauli-type gates like `X(0)*X(1)` but not mixed ones.